### PR TITLE
System table and function reference update

### PR DESCRIPTION
**Pull request by fulghum**
Documentation updates for system tables and sql functions:
* reordered sections to match index order
* updated `dolt_commit_diff_$tablename`, `dolt_diff_$tablename`, and `dolt_history_$tablename` system tables for new behavior where schema comes from current branch and does not disambiguate column names and better explain differences in diff tables
* various small typo fixes, links to referenced functions, etc

Rendered views:
* [dolt-system-tables.md](https://github.com/dolthub/docs/blob/bff6ae89ea4eb42ed42538201baa5066f230df1c/content/reference/sql/dolt-system-tables.md)
* [dolt-sql-functions.md](https://github.com/dolthub/docs/blob/bff6ae89ea4eb42ed42538201baa5066f230df1c/content/reference/sql/dolt-sql-functions.md)

### CODE REVIEW

File: content/reference/sql/dolt-sql-functions.md [hunk 1]
```diff
@@ -4,21 +4,24 @@ title: Dolt SQL Functions
...
+  * [dolt\_fetch()](#dolt_fetch)
+  * [dolt\_merge()](#dolt_merge)
+  * [dolt\_pull()](#dolt_pull)
+  * [dolt\_push()](#dolt_push)
+  * [dolt\_reset()](#dolt_reset)
+  
+* [Dolt Commit Metadata SQL Functions](#dolt_commit_metadata_sql_functions)
+  * [dolt\_merge\_base()](#dolt_merge_base)
+  * [hashof()](#hashof)
+  * [detached head functions](#detached_head_functions)
```
Comment (zachmu) [inline content/reference/sql/dolt-sql-functions.md]: These are now dead, can you confirm @andrew-wm-arthur ?

File: content/reference/sql/dolt-sql-functions.md [hunk 2]
```diff
@@ -382,10 +391,10 @@ SELECT DOLT_PULL('origin');
 SELECT * FROM dolt_log LIMIT 5;
 ```
 
-# Other Dolt SQL functions
+# Dolt Commit Metadata SQL Functions
 
 Dolt also provides various SQL functions for dolt-specific
-functionality in SQL that have no command line equivalent.
+functionality that has no command line equivalent.
```
Comment (zachmu) [inline content/reference/sql/dolt-sql-functions.md]: have is correct (refers to "functions", not "functionality")

File: content/reference/sql/dolt-sql-functions.md [hunk 3]
```diff
@@ -397,22 +406,22 @@ Consider the following branch structure:
...
 ```
 
 ## `HASHOF()`
 
 The HASHOF function returns the commit hash of a branch,
-e.g. `HASHOF("master")`.
+e.g. `HASHOF("main")`.
 
-## Functions for working with detached heads
+## Detached Head Functions
```
Comment (zachmu) [inline content/reference/sql/dolt-sql-functions.md]: Kill plz

Comment (zachmu) [reply content/reference/sql/dolt-sql-functions.md]: Also search for refs to these docs, there are other places that will need to be updated

File: content/reference/sql/dolt-system-tables.md [hunk 4]
```diff
@@ -108,47 +108,357 @@ INSERT INTO dolt_branches (name, hash)
...
+| commit_date       | datetime |
+| committer         | text     |
+| email             | text     |
+| message           | text     |
+| primary key cols  |          |
++-------------------+----------+
+```
+
+The remaining columns are dependent on the schema of the user table.
+Every column from the primary key of your table will be included in the `dolt_blame_$tablename` system table.
```
Comment (zachmu) [inline content/reference/sql/dolt-system-tables.md:357]: Mention behavior for unkeyed tables?

File: content/reference/sql/dolt-system-tables.md [hunk 5]
```diff
@@ -108,47 +108,357 @@ INSERT INTO dolt_branches (name, hash)
...
+| id  | commit                           | commit_date                       | committer       | email             | message                       |
++-----+----------------------------------+-----------------------------------+-----------------+-------------------+-------------------------------+
+| 1   | gift4cdu4m0daedgppu8m3uiuh8sovc8 | 2022-02-22 20:05:08.881 +0000 UTC | Thomas Foolery, | foolery@email.com | updating display config value |
+| 2   | 30c2qqv3u6mvfsd11g0t1ejk0j974f71 | 2022-02-22 20:05:09.14 +0000 UTC  | Harry Wombat,   | wombat@email.com  | switching to file encryption  |
+| 3   | s15jrjbtg1mq5sfmekpgdomijcr4jsq0 | 2022-02-22 20:05:09.265 +0000 UTC | Johnny Moolah,  | johnny@moolah.com | adding new config for format  |
+| 4   | s15jrjbtg1mq5sfmekpgdomijcr4jsq0 | 2022-02-22 20:05:09.265 +0000 UTC | Johnny Moolah,  | johnny@moolah.com | adding new config for format  |
++-----+----------------------------------+-----------------------------------+-----------------+-------------------+-------------------------------+
+```
+
```
Comment (zachmu) [inline content/reference/sql/dolt-system-tables.md:395]: Extra blank line here and in other sections

**Review by zachmu — APPROVED** [review]
LGTM!

Main comment is to kill off the detached head info, since it's no longer supported. Could do that in a followup.

File: content/reference/sql/dolt-system-tables.md (hunk 5 above)
Comment (fulghum) [reply content/reference/sql/dolt-system-tables.md:395]: This doesn't seem to render in the HTML output and makes the section breaks a little easier to see in the source, but if you have concerns I'm happy to rip them out.

**Pull request merged**